src/main/scala/mesosphere/marathon/api/akkahttp/v2/GroupsController.scala

```diff
@@ -76,4 +96,29 @@ class GroupsController(electionService: ElectionService, infoService: GroupInfoS
     }
   }
   // format: On
+
+  def authorizationSelectors(implicit identity: Identity): GroupInfoService.Selectors = {
+    GroupInfoService.Selectors(
+      AppHelpers.authzSelector,
+      PodsResource.authzSelector,
+      authzSelector)
+  }
+
+  import mesosphere.marathon.api.v2.InfoEmbedResolver._
+  /**
+    * For backward compatibility, we embed always apps, pods, and groups if nothing is specified.
+    */
+  val defaultEmbeds = Set(EmbedApps, EmbedPods, EmbedGroups)
+  def extractEmbeds: Directive1[(Set[AppInfo.Embed], Set[GroupInfo.Embed])] = {
+    parameter('embed.*).tflatMap {
+      case Tuple1(embeds) => provide(resolveAppGroup(if (embeds.isEmpty) defaultEmbeds else embeds.toSet))
+    }
+  }
+}
+
+object GroupsController {
```

Comment:
What is the motivation for this object?

Reply:
I took this whole thing from Resource class, I guess it does not have to be an object